### ISSUE TITLE
doasedit: copy to temporary fails if the path is absolute

### ISSUE THREAD

**Issue by yshybeka** [slicer69/doas#128]
I'm running FreeBSD-14. The issue can be seen in the following example:

```
% sh -c 'echo `dirname -- "$1"`/$1' -- /bin/sh /etc/rc.conf
/etc//etc/rc.conf
```

and it cannot find `/etc//etc/rc.conf`. on the other hand `cp -- ...` works fine at least on FreeBSD:

```
% sh -c 'if ! cp -- "$1" `mktemp -t /tmp`; then echo FAILED; fi' -- /bin/sh /etc/rc.conf
```

```
% sh -c 'if ! cp -- "$1" `mktemp -t /tmp`; then echo FAILED; fi' -- /bin/sh /etc/rc.conf.NONEXISTENT
cp: /etc/rc.conf.NONEXISTENT: No such file or directory
FAILED
```

**Comment by slicer69** [comment]
This has been fixed in 6.3p12. Thanks for the report.

**Comment by yshybeka** [comment]
thank you!
